tests/robot-tests/scripts/create_snapshots.py

```diff
@@ -219,18 +221,57 @@ def _create_methodologies_snapshot(self) -> str:
             themes.append(theme)
         return json.dumps(themes, sort_keys=True, indent=2)
 
-    def validate_snapshots(self):
+    def _build_slack_blocks(self, messages: List[str]) -> List | None:
+        # if length of messages is 0, then no errors were found
+        blocks = []
+        if len(messages) > 0:
+            # PR is raised due to snapshot mismatch
+            blocks.append(
+                {
+                    "type": "header",
+                    "text": {
+                        "type": "plain_text",
+                        "text": "Snapshots do not match . See pull request for more details",
+                    },
+                }
+            )
+
+            blocks.append(
+                {
+                    "type": "actions",
+                    "elements": [
+                        {
+                            "type": "button",
+                            "style": "danger",
+                            "text": {"type": "plain_text", "text": "View pull request"},
+                            "url": "https://github.com/dfe-analytical-services/explore-education-statistics/pulls/dfe-sdt",
+                        }
+                    ],
+                }
+            )
+        return blocks if len(blocks) > 0 else None
+
+    def validate_snapshots(self) -> None:
+        messages: List[str] = []
+
         for name in self.snapshots:
             if self._validate_snapshot(name):
                 print(f"Snapshot {name} matches for {self.public_url}")
             else:
                 print(f"Snapshot {name} does not match for {self.public_url}")
-                if self.slack_webhook_url is not None:
-                    self._send_slack_notification(f"Snapshot {name} does not match for {self.public_url}")
-        if self.slack_webhook_url is not None:
-            self._send_slack_notification(f"Snapshot test workflow successfully completed for {self.public_url} ✅")
+                messages.append(f"Snapshot `{name}` does not match for {self.public_url}")
```

Comment:
I think `messages` could be a boolean - I don't believe the string appended to the `messages` list is ever being used.

So instead this could be something like
```
failed: bool = False

for name in self.snapshots:
  if self._validate_snapshot(name):
    ...
  else:
    print(...)
    failed = True

if failed and self.slack_webhook_url:
  ...
```